File: MAP/Lab 2/ToyInterpreter/src/model/statements/AssignStmt.java
package model.statements;

import model.adts.MyIDictionary;
import model.adts.MyIStack;
import model.expressions.IExp;
import model.state.PrgState;

public class AssignStmt implements IStmt{
    private String id;
    private IExp exp;

    public AssignStmt(String id, IExp exp) {
        this.id = id;
        this.exp = exp;
    }

    @Override
    public String toString() {
        return id + "="+ exp.toString();
    }

    public PrgState execute(PrgState state) throws Exception {
        MyIStack<IStmt> stk=state.getExeStack();
        MyIDictionary<String,Integer> symTbl= state.getSymTable();
        MyIDictionary<Integer, Integer> heapTbl = state.getHeap();
        int val = exp.eval(symTbl, heapTbl);

        if(symTbl.find(id))
            symTbl.update(id, val);
        else
            symTbl.insert(id,val);

        return null;
    }

}

File: MAP/Lab10/ConsoleApp1/ConsoleApp1/Model/statements/AssignStmt.cs
using ConsoleApp1.Model.state;
using ConsoleApp1.Model.statements;
using System;
using System.Collections.Generic;
using System.Text;
using ToyInterpreter.Model.expressions;

namespace ToyInterpreter.Model.statements
{
    class AssignStmt: IStmt
    {
        private String id;
        private IExp exp;

        public AssignStmt(String id, IExp exp)
        {
            this.id = id;
            this.exp = exp;
        }

        public PrgState execute(PrgState state)
        {
            Stack<IStmt> stk = state.getExeStack();
            Dictionary<String, Int64> symTbl = state.getSymTable();
            int val = exp.eval(symTbl);

            symTbl.Add(id, val);
            return state;
        }

        public override String ToString()
        {
            return id + "=" + exp.ToString();
        }  
    }
}
